backend/src/main/java/com/redhat/cloud/notifications/db/EventResources.java

```diff
@@ -116,7 +117,7 @@
     private static String addHqlConditions(String hql, Set<UUID> bundleIds, Set<UUID> appIds, String eventTypeDisplayName,
                                            LocalDate startDate, LocalDate endDate, Set<EndpointType> endpointTypes, Set<Boolean> invocationResults) {
         if (bundleIds != null && !bundleIds.isEmpty()) {
-            hql += " AND b.id IN (:bundleIds)";
+            hql += " AND a.bundle.id IN (:bundleIds)";
```

Comment:
I suppose this adds the JOIN on demand?

Reply:
No the join with `bundles` is gone for good.

Old query:
```
[...]
 inner join
        bundles bundle3_
            on applicatio2_.bundle_id=bundle3_.id
[...]
```
New query:
```
[...]
 applicatio2_.bundle_id in (
                $2
            )
[...]
```

Reply:
nice!

Reply:
Hmm not so nice, the tests showed me something wrong because of the ordering 😄 This needs more work anyway!

Reply:
@josejulio Do you think this is needed?
https://github.com/RedHatInsights/notifications-backend/blob/bcb91e750c40b0128300ce89e41699797bc3c846/backend/src/main/java/com/redhat/cloud/notifications/db/EventResources.java#L179-L184

The frontend only sorts on the creation time which makes sense.

Reply:
I'm sure we will eventually be asked (if not already in the mocks) to sort by any column.

Reply:
No longer relevant, my approach to fix the response time has changed.